cypress/integration/hidden-features.spec.ts

```diff
@@ -0,0 +1,221 @@
+        }
+      });
+  });
+  cy.get('[data-cyId=onboarding-nextButton]').should('be.disabled');
+  cy.get('[data-cyId=onboarding-error]').scrollIntoView().should('be.visible');
+
+  // set Graduation year to 2022
+  cy.get('[data-cyId=onboarding-dropdown]').eq(1).click();
+  cy.get('[data-cyId=onboarding-dropdownItem]').each($el => {
+    cy.wrap($el)
+      .invoke('text')
+      .then(text => {
+        if (text.includes('2022')) {
+          cy.wrap($el).click();
+        }
+      });
+  });
+  cy.get('[data-cyId=onboarding-nextButton]').should('be.disabled');
+  cy.get('[data-cyId=onboarding-error]').scrollIntoView().should('be.visible');
+
+  // set to Engineering college
+  cy.get('[data-cyId=onboarding-dropdown]').eq(2).click();
+  cy.get('[data-cyId=onboarding-dropdownItem]').each($el => {
+    cy.wrap($el)
+      .invoke('text')
+      .then(text => {
+        if (text.includes('Engineering')) {
+          cy.wrap($el).click();
+        }
+      });
+  });
+
+  // next button can be clicked and error not visible now that every field has been selected
+  cy.get('[data-cyId=onboarding-nextButton]').click();
+  cy.get('[data-cyId=onboarding-error]').should('not.exist');
+  cy.get('[data-cyId=onboarding-nextButton]').click();
+
+  // confirm 2018, 2022, and engineering are selected on the review screen
+  cy.get('[data-cyId=onboarding-entranceYear]').contains('2018');
+  cy.get('[data-cyId=onboarding-gradYear]').contains('2022');
+  cy.get('[data-cyId=onboarding-college]').contains('Engineering');
+  cy.get('[data-cyId=onboarding-finishButton]').click();
+});
+
+// Test to confirm that the new user walkthrough works as expected
+// Click through the initial explanation, then the 4 following steps, and finally the finishing page
+it('Click through the walkthrough tour', () => {
+  cy.get('[data-cyId=tour]').should('be.visible');
+  cy.get('[data-cyId=tour-startButton]').click();
+
+  // click through each step of the walkthrough, and confirm the tooltips are visible
+  // note that the introjs DOM elements come from the package, so cyID attributes cannot be used
+  cy.get('.tourStep1').should('be.visible');
+  cy.get('.introjs-nextbutton').click();
+
+  cy.get('.tourStep2').should('be.visible');
+  cy.get('.introjs-nextbutton').click();
+
+  cy.get('.tourStep3').should('be.visible');
+  cy.get('.introjs-nextbutton').click();
+
+  cy.get('.tourStep4').should('be.visible');
+  cy.get('.introjs-nextbutton').click();
+
+  // confirm the final page is visible and the tour can be finished
+  cy.get('[data-cyId=tour]').should('be.visible');
+  cy.get('[data-cyId=tour-startButton]').click();
+});
+
+// Test to confirm that only one "+New Semester" button shows up at once
+// Due to front-end styling issues, there are two different buttons in different components
+// Only one should be visible at a time (from semesterView if there are 0 semesters, from semester otherwise)
+it('Test that only one semester button shows', () => {
```

Comment:
Is it possible to have this test be perpetual (i.e. throughout the other tests, ensure the invariant holds) rather than for one specific set of steps?

Reply:
Yes! I think I can add an `afterEach` hook to run it after every test that we have. It will require reworking things a bit, so the test only checks for buttons and does not actually click anything, and also needs to ensure that the site is in a state where this can be checked. Adding it to my other item!

Reply:
> and also needs to ensure that the site is in a state where this can be checked.

I think this can just be done by making the invariant "there is max 1 button" instead of "there is exactly 1 button". So if no buttons exist, the test passes.

"There is exactly 1 button" is not really related to the bug itself and can be another test if we want.